pkg/providers/v1/aws.go

```diff
@@ -2138,14 +2193,128 @@ func (c *Cloud) buildNLBHealthCheckConfiguration(svc *v1.Service) (healthCheckCo
 	return hc, nil
 }
 
+// ensureNLBSecurityGroup ensures the NLB security group is created and configured
+// based on the current NLB state and configuration mode.
+//
+// This function handles three security group modes:
+//  1. BYO (Bring Your Own): Uses security groups from annotations or global config
+//  2. Managed: Creates and manages security groups automatically when enabled
+//  3. Disabled: Returns empty security group list when not in managed mode
```

Comment:
This documentation is outdated from the latest change. I suggest to simplify:
```suggestion
// ensureNLBSecurityGroup ensures the NLB security group is created and configured
// based on the current NLB state and configuration mode, and and ensure limitations
// are clearly reported by users, such as no BYO SG support.
```